### PR TITLE
Cherry pick Omnitrace docs refactoring (#353)

### PR DESCRIPTION
**Pull request by samjwu**
Cherry pick of Omnitrace docs refactoring (#353)

### CODE REVIEW

**Review by peterjunpark — APPROVED** [review]

File: docs/reference/development-guide.rst [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+* Not installed or exported outside of the build tree.
+
+Binary library: `source/lib/binary <https://github.com/ROCm/omnitrace/tree/main/source/lib/binary>`_
+--------------------------------------------------------------------------------------------------------------------------------
+
+* Static PIC library with functionality for reading/analyzing binary info.
+* Mostly used by the causal profiling sections of ``libomnitrace``.
+* Not installed or exported outside of the build tree.
+
+libomnitrace: `source/lib/omnitrace <https://github.com/ROCm/omnitrace/tree/main/source/lib/omnitrace>`_
```
Comment (dgaliffiAMD) [inline docs/reference/development-guide.rst:92]: Should these links be updated to point to the release branch? Or changed to use relative links?

Comment (amd-jnovotny) [reply docs/reference/development-guide.rst:92]: Right now, they seem to go to the "main" branch in GitHub. Users can select a branch from there. But I'll ask around about this to find out how this is handled elsewhere.

Comment (samjwu) [reply docs/reference/development-guide.rst:92]: Currently only links to other documentation via intersphinx (https://rocm.docs.amd.com/projects/rocm-docs-core/en/latest/user_guide/linking.html) are able to take the branch as a parameter (ie: 6.0 links to other 6.0)

Doing the same for links to GitHub is still an in-progress feature

Comment (dgaliffiAMD) [reply docs/reference/development-guide.rst:92]: In that case, should we change to <https://github.com/ROCm/omnitrace/tree/release/rocm-rel-6.2/source/lib/omnitrace>.
And similarly change other links from `/tree/main` to `tree/release/rocm-rel-6.2`.

Comment (amd-jnovotny) [reply docs/reference/development-guide.rst:92]: The one downside of that is that we have to remember to update the links with every release. So if we don't update the links for 6.3, then they will point to 6.2. But I'm open to discussion on this.

**Review by amd-jnovotny — APPROVED** [review]
looks good

**Comment by samjwu** [comment]
merging this. if any additional changes need to be made, a new PR can be made to address

**Pull request merged**